packages/services/src/vckit.service.ts

```diff
@@ -43,18 +56,68 @@ export const issueVC = async ({
   type,
   issuer,
   credentialSubject,
+  credentialStatus,
   restOfVC,
   vcKitAPIUrl,
 }: IVcKitIssueVC): Promise<VerifiableCredential> => {
-  const body = constructCredentialObject({ context, type, issuer, credentialSubject, ...restOfVC });
-  privateAPI.setBearerTokenAuthorizationHeaders(appConfig.defaultVerificationServiceLink.apiKey ?? '');
+  const apiKey = appConfig.defaultVerificationServiceLink.apiKey ?? '';
+  let _credentialStatus = credentialStatus ? { ...credentialStatus } : null;
+
+  // issue credential status if not provided
+  if (!_credentialStatus) {
+    // issue credential status
+    _credentialStatus = await issueCredentialStatus({
+      host: new URL(vcKitAPIUrl).origin, // example: https://api.vc.example.com
+      apiKey,
+      bitstringStatusIssuer: issuer,
+    });
+  }
+
+  // issue vc
+  const body = constructCredentialObject({ context, type, issuer, credentialSubject, credentialStatus: _credentialStatus, ...restOfVC });
+  privateAPI.setBearerTokenAuthorizationHeaders(apiKey);
   const { verifiableCredential } = await privateAPI.post<VerifiableCredential>(
     `${vcKitAPIUrl}/credentials/issue`,
     body,
   );
   return verifiableCredential;
 };
 
+/**
+ * Issue credential status for the vc
+ * @param host - host for issuing the credential status
+ * @param apiKey - api key for issuing the credential status
+ * @param statusPurpose - purpose for the credential status
+ * @param bitstringStatusIssuer - issuer for the credential status
+ * @returns CredentialStatusReference
+ * 
+ * @example
+ * const host = 'https://api.vc.example.com';
+ * const apiKey = 'api-key';
+ * const statusPurpose = 'revocation';
+ * const bitstringStatusIssuer = 'did:example:123';
+ * const credentialStatus = await issueCredentialStatus({ host, statusPurpose, bitstringStatusIssuer });
+ */
+export const issueCredentialStatus = async (args: IArgIssueCredentialStatus): Promise<CredentialStatusReference> => {
+  if (!args.host) {
+    throw new Error('Error issuing credential status: Host is required');
+  }
+  if (!args.apiKey) {
+    throw new Error('Error issuing credential status: API Key is required');
+  }
+  if (!args.bitstringStatusIssuer) {
+    throw new Error('Error issuing credential status: Bitstring Status Issuer is required');
```

Comment:
I am just curious about why don't we add `apiKey` and `bitstringStatusIssuer` to `IArgIssueCredentialStatus`. Now, I am seeing we only have the required host but we check apiKey in here.

Reply:
Good point! I missed that in the IArgIssueCredentialStatus interface. Thanks for bringing it up.